app/Services/InviteService.php

```diff
@@ -2,44 +2,164 @@
+use App\Mail\InviteMail;
 use App\Models\InviteRole;
+
+use Illuminate\Support\Collection;
+use Tymon\JWTAuth\Facades\JWTAuth;
+use App\Repositories\RoleRepository;
+use App\Repositories\UserRepository;
+use Illuminate\Support\Facades\Mail;
+use App\Repositories\StaffRepository;
+use Tymon\JWTAuth\Facades\JWTFactory;
 use App\Repositories\InviteRepository;
-use Carbon\Carbon;
+use App\Repositories\StudentRepository;
 use Illuminate\Validation\ValidationException;
-use Illuminate\Support\Collection;
+
 
 class InviteService
 {
+
+    private $staffRepository, $roleRepository, $studentRepository;
+
+    public function __construct(RoleRepository $roleRepository, StaffRepository $staffRepository, StudentRepository $studentRepository)
+    {
+        $this->roleRepository = $roleRepository;
+        $this->staffRepository = $staffRepository;
+        $this->studentRepository = $studentRepository;
+    }
+
     public function repo()
     {
         return app(InviteRepository::class);
     }
 
-    public function create($opts) {
+    public function create($opts)
+    {
         $invite = $this->repo()->create($opts);
 
         if (isset($opts['role_id'])) {
             $role_id = $opts['role_id'];
             $invite->roles()->create([
-                'role_id' => $role_id
+                'role_id' => $role_id,
             ]);
-        }
-        else if (isset($opts['roles'])) {
+        } else if (isset($opts['roles'])) {
             $roleIds = new Collection($opts['roles']);
             $roleIds->map(function ($data) use ($invite) {
                 $invite->roles()->create([
                     'role_id' => $data['role_id'],
-                    'extras' => isset($data['extras']) ? $data['extras'] : null
+                    'extras' => isset($data['extras']) ? $data['extras'] : null,
                 ]);
             });
-        }
-        else {
+        } else {
             throw ValidationException::withMessages([
                 'role_id' => 'should exist if "roles" are absent"',
-                'roles' => 'should exist if "role_id" is absent"'
+                'roles' => 'should exist if "role_id" is absent"',
             ]);
         }
         $invite->load('roles');
+        $this->sendMail($invite);
         return $invite;
     }
-}
\ No newline at end of file
+
+    public function resolve($id)
+    {
+        $invite = $this->repo()->single($id);
+        $roles = $invite->roles()->get();
+        $user = auth()->user();
+        $school_id = $invite->school_id;
+
+        foreach ($roles as $_role) {
+            $role_id = $_role->role_id;
+            $extras = json_decode($_role->extras);
+            $role = $this->roleRepository->single($role_id);
+
+            if ($role->name == Role::ADMIN) {
+                $admin = $user->roles()->attach([
+                    $role->id => [
+                        'school_id' => $school_id,
+                    ],
+                ]);
+                return $admin;
```

Comment:
If you return `$admin` here, the foreach loop would not finish. Also, on line 83, an admin does not need to have a `school_id`.

Reply:
You should always return the user.